### PR TITLE
Fix v1beta2: Use UpdateStatus to update subresource

### PR DESCRIPTION
**Pull request by akhurana001**
@liyinan926 here is a quick fix for https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/644 .  Haven't  extensively tested but the sample examples run fine with this change . Also didn't test the `ScheduledSparkApplication`.

I also ran into `validation failure list:
status.terminationTime in body must be of type string: "null"` errors.  Haven't looked much into this but removing the validation did fix it.  I think K8s also started supporting `nullable: true` in 1.14 which should work as well.

### CODE REVIEW

File: manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml [hunk 1]
```diff
@@ -2563,9 +2563,6 @@ spec:
               type: integer
             submissionID:
               type: string
-            terminationTime:
```
Comment (liyinan926) [inline manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Why this change?

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Didn't see your message above. Removing the check is fine for now before we can add `nullable: true`. I think we need to remove this check for all `metav1.Time` field in both `SparkApplication` and `ScheduledSparkApplication`.

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Actually a quick fix is to add `omitempty`.

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: It seems that `controller-gen` supports the `+nullable` tag. Can we try that?

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Looks like `nullable` support is only available beginning 1.14. So I guess for now we need to remove this check. Actually I don't even think validation on `status` makes any sense for now. Let's remove validation for `status` all together.

Comment (akhurana001) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Yes, nullable support starts with 1.14.  Agreed, we can remove the validations for status. Let me do that

**Comment by liyinan926** [comment]
@khogeland FYI not sure if you ran into the issue this PR tries to address or not.

File: manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml (hunk 1 above)
Comment (akhurana001) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: I did remove the validations manually. I don't see any usable marker in https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/markers/validation.go which would achieve the same ? Let me know if there is one.

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: See https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/markers/validation.go#L163.

Comment (akhurana001) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Not sure on https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/markers/validation.go#L163 . I think this is the same `+nullable` tag which sets `nullable:true` in the CRD. It will let the status field be null but  if the `status` field is not null, it will still validate the types and we will run into the same `terminationTime` issue. Maybe I am missing something here.

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Yes, that's right. But for `status` as a whole, it's type is `object`, and I think they fixed the issue with empty `status`. We can still apply `+nullable` to those `metav1.Time` fields to solve the issue.

Comment (akhurana001) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: `+nullable` will need 1.14+ support as we discussed before. I can do that in another PR if you think that's the right way.

Comment (liyinan926) [reply manifest/crds/sparkoperator.k8s.io_sparkapplications.yaml]: Yes, I meant for now we just remove validation for status, and in the future, we should add it back and use `+nullable` on those time fields.
